### ISSUE TITLE
curve_fit: "LinAlgError: SVD did not converge"

### ISSUE THREAD

**Issue by arunsinp** [scipy/scipy#18965]
I have data frame `df` with the following columns:

|    X     |         Y      |   delta_t       |
|-------|----------|-------------|
|     .      |         .       |          .           |
|     .      |         .       |          .           |
|     .      |         .       |          .           |
|     .      |         .       |          .           |
|     .      |         .       |          .           |

and I am trying to fit the exponential function:  Y/X = a _x_ Exp(-(b_1 - b_2) _x_ c _x_ delta_t).

With a given initial value as shown in the code (given below). I am getting an issue `"LinAlgError: SVD did not converge"`. 

How to overcome this issue? 

```python

-------------------- something about the variables -------------------------
`df['Y'].describe()`:
count    3537.000000
mean       19.985690
std        22.486308
min         0.582168
25%         5.446700
50%        10.997200
75%        25.789900
max       121.417000
---------------------------------------------------------------------------
`df['X'].describe()`:
count    3537.000000
mean        5.686478
std         5.564841
min         0.265318
25%         1.928670
50%         3.581990
75%         6.938480
max        26.451900
---------------------------------------------------------------------------
`df['delta_t'].describe()`:
count    1318.000000
mean       10.666609
std         5.854462
min        -6.644779
25%         6.354654
50%         9.811106
75%        14.335943
max        46.353073
---------------------------------------------------------------------------

```

-----------------------------------------------------------------------------------------------------------
 |                                                                   The Code is as follows                                                                           |
-----------------------------------------------------------------------------------------------------------
```python
b_2 = 0.83e-12
c = 2.5e6

df.dropna(subset=['Y / X'], inplace=True)

Q1 = df['Y / X'].quantile(0.25)
Q3 = df['Y / X'].quantile(0.75)
IQR = Q3 - Q1
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR
dfsum_copy2 = df[(df['Y / X'] >= lower_bound) & (df['Y / X'] <= upper_bound)]

x_data = df['c']
y_data = df['Y / X']

def fitting_function(delta_t, a, b_2):
    return a * np.exp(-(b_1 - b_2) * c* delta_t)

x_data_normalized = (x_data - x_data.mean()) / x_data.std()
y_data_normalized = (y_data - y_data.mean()) / y_data.std()

initial_guess = [0.22, 1.25e-12]  
fit_params, pcov = curve_fit(fitting_function, x_data_normalized, y_data_normalized, p0=initial_guess, method='lm')

a_fit, b_1_fit = fit_params

x_curve = np.linspace(min(x_data), max(x_data), 1000)
y_curve = fitting_function(x_curve, a_fit, b_1_fit)

plt.scatter(x_data, y_data, label='Data')
plt.plot(x_curve, y_curve, 'r-', label='Fitted Curve')
plt.xlabel('delta_t')
plt.ylabel('Y/ X')
plt.legend()
plt.grid(True)
plt.show()

a_fit, b_1_fit = fit_params
print("Fitted Parameters:")
print("a =", a_fit)
print("b_1 =", b_1)`
```

**Comment by arunsinp** [comment]
np.linalg.cond(pcov)

shows the following error:

---------------------------------------------------------------------------
LinAlgError                               Traceback (most recent call last)
Cell In[202], line 1
----> 1 np.linalg.cond(pcov)

File <__array_function__ internals>:200, in cond(*args, **kwargs)

File [/usr/local/lib/python3.11/site-packages/numpy/linalg/linalg.py:1747](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.11/site-packages/numpy/linalg/linalg.py:1747), in cond(x, p)
   1745     raise LinAlgError("cond is not defined on empty arrays")
   1746 if p is None or p == 2 or p == -2:
-> 1747     s = svd(x, compute_uv=False)
   1748     with errstate(all='ignore'):
   1749         if p == -2:

File <__array_function__ internals>:200, in svd(*args, **kwargs)

File [/usr/local/lib/python3.11/site-packages/numpy/linalg/linalg.py:1654](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.11/site-packages/numpy/linalg/linalg.py:1654), in svd(a, full_matrices, compute_uv, hermitian)
   1651     gufunc = _umath_linalg.svd_n
   1653 signature = 'D->d' if isComplexType(t) else 'd->d'
-> 1654 s = gufunc(a, signature=signature, extobj=extobj)
   1655 s = s.astype(_realType(result_t), copy=False)
   1656 return s

File [/usr/local/lib/python3.11/site-packages/numpy/linalg/linalg.py:98](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.11/site-packages/numpy/linalg/linalg.py:98), in _raise_linalgerror_svd_nonconvergence(err, flag)
     97 def _raise_linalgerror_svd_nonconvergence(err, flag):
---> 98     raise LinAlgError("SVD did not converge")

LinAlgError: SVD did not converge

**Comment by tupui** [comment]
Hi @arunsinp, thank you for reporting.

I am afraid it is going to be difficult to help more without a minimal reproducible code we can run. (It would have been helpful if you had followed our template.)

What did you try? Did you read the documentation of the function and tried some parameters? 

https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html

Based on the information you provided, it's hard to tell if this is a user question or if there is an issue. Note that for user related question, the issue tracker is not the right place and we invite such questions either on Slack or StackOverflow. See https://scipy.org/community

**Comment by arunsinp** [comment]
Hey Pamphile, Thanks for responding. I read the document and followed step by step. Everything worked perfectly but my code failed the convergence test. Sorry, I did not know that user-related questions are not raised here. I have already asked a question on Stackflow (https://stackoverflow.com/questions/76769454/curve-fit-linalgerror-svd-did-not-converge).

Currently, I am working with some atmospheric datasets and I am trying to fit an exponential function and get some of the unknowns.  I used curve_fit:
```
initial_guess = [0.22, 1.25e-12]  
fit_params, pcov = curve_fit(fitting_function, x_data_normalized, y_data_normalized, p0=initial_guess, method='lm')
```
and when I calculate the condition number of the covariance matrix (pcov) obtained from the curve_fit function, I am getting an error. 

Thanks for letting me know. I will keep checking Stackflow response section and try to find some resources for my future work.

**Comment by tupui** [comment]
Did you check that `x_data_normalized, y_data_normalized` do not have inf/NaNs?

**Comment by arunsinp** [comment]
I have already checked...there are no inf/NaNs values present. I think I did some mistakes in providing the initial values for the parameters and also I need to recheck the removed outliers. When I tried to remove outliers by IQR method the number of data points reduced significantly. I am now trying to remove them using the z-score method. Which method should be used to remove the outliers? In my case, there are very few outliers but they have huge values so they are significantly changing the statistics of the analysis. 

Sorry for asking such a stupid question.

Thanks
